ps3netsrv: Update to v1.47.46e with ps3netsrv 20240210a

Hirador opened the pull request:
Update to v1.47.46e with ps3netsrv 20240210a for DSM's versions: DSM-6.2.4 -> DSM-7.1

## Description

1. Updated the variables found on `spksrc/spk/ps3netsrv` and `spksrc/cross/ps3netsrv` to match the current version (v1.47.46e with ps3netsrv 20240210a) of aldostools's ps3netsrv
2. made "make all-supported"
3. created .spk for my Architecture (arch-armada370)
4. Made some tests with the updated version of the ps3netsrv
5. Cleaned everything up for the pull request (make clean on both spksrc/spk/ps3netsrv and spksrc/cross/ps3netsrv)

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relevant tags.-->
- [ ] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

Hirador commented:
Can everything be merged in the next commit now that I have created a dedicated branch?

hgy59 commented:
> @hgy59 Can everything be merged in the next commit now that I have created a dedicated branch?

I can see an official release 1.47.46 in https://github.com/aldostools/webMAN-MOD/tags
it was created on Mar 21 2024.
Probably only the changelog is outdated?
`CHANGELOG = "Update to v1.47.46b with ps3netsrv 20240210."`

Hirador commented:
@hgy59 when I was working on this commit the version was 1.47.46b, even if the code should be the same the tar.gz could have different md5, I'll make some checks and see if everything it's okay. Just to be as precise as possible I would in any case update the comment and description, if so I think that this pull request should be closed and recreated again, right?

mreid-tt commented:
@Hirador, opening a new PR for version changes isn't always necessary. You can simply update the PR description and the change log to reflect the current versions being committed. This approach not only saves time but also makes it easier for reviewers to track minor changes like version updates, rather than reviewing a new PR from scratch.

Hirador commented:
@mreid-tt I've made all the check, apparently the md5 is the same, i've changed the changelog with the updated values and changed the pull request title, i think everything is good to go now

Hirador commented:
@mreid-tt @hgy59 I've made the required changes, please check everything out to be sure that i've made them properly (May have missed the DSM version check, don't know if it is mandatory in my case). If everything is fine I'll test the .spk on my DiskStation DS414slim (Marvell Armada 370) with DSM 7.1.1 later today, if you could test it on DSM 5.X , 6.X, and SRM 1.X it would be great!

hgy59 commented on spk/ps3netsrv/src/service-setup.sh:
```diff
@@ -7,6 +7,6 @@ export CFG_FILE PID_FILE SERVICE_PORT
 {
     if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
-        sed -i -e "s|@wizard_dir@|${wizard_data_volume}/${wizard_data_directory}|g" ${CFG_FILE}
+        sed -i -e "s|@wizard_dir@|${SHARE_PATH}|g" ${CFG_FILE}
```
suggest that you must apply the changed placeholder in ps3netsrv.conf file

```suggestion
        sed -i -e "s|@share_path@|${SHARE_PATH}|g" ${CFG_FILE}
```

Hirador replied:
Oh my bad, sorry.

Fixed and committed

Hirador commented:
@hgy59 @mreid-tt Made a few tests on DSM 7.1.1, everything seems to be working fine, i think we may be ready to merge after your further tests!

Hirador commented:
@hgy59 did you manage to check up everything so to proceed in merging?

hgy59 commented:
The migration of `installer-variables` isn't correct. The upgrade might be missing something.

With the default share "PS3" the etc/installer-variables file looks like this:

former package version:
```
SHARE_PATH=PS3
```

new installation of new version:
```
SHARE_PATH=/volume1/PS3
SHARE_NAME=PS3
```

upgrade from former to new version:
```
SHARE_PATH=PS3
SHARE_NAME=PS3
```

As we can see, the `SHARE_PATH` variable is not migrated to the full name `/volume1/PS3`

AFAICR this was fixed by @mreid-tt for some other packages...

hgy59 commented:
package upgrade on DSM 6 fails ❗

The issue with etc/installer-variable reported above occurs with DSM 7 (tested with DSM 7.2.1-69057 Update 5).

On DSM 6.2.4 the package upgrade from former version fails. (on DSM 6 there is no "Repair" button provided).

The /var/log/messages file reports an error `Failed to acquire resource after install ps3netsrv`
```
2024-06-29T11:59:52+02:00 ##### synoscgi_SYNO.Core.Package_2_list[15440]: list.cpp:329 SYNO.Core.Package.list rollback to direct call
2024-06-29T11:59:58+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[17667]: resource_api.cpp:289 Release service-cfg for ps3netsrv when 0x0002 (done)
2024-06-29T11:59:58+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[17673]: resource_api.cpp:289 Release port-config for ps3netsrv when 0x0002 (done)
2024-06-29T11:59:59+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[17730]: resource_api.cpp:190 Acquire data-share for ps3netsrv when 0x0002 (done)
2024-06-29T11:59:59+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[17799]: resource_api.cpp:190 Acquire service-cfg for ps3netsrv when 0x0002 (done)
2024-06-29T11:59:59+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[17799]: resource_api.cpp:190 Acquire data-share for ps3netsrv when 0x0002 (fail)
2024-06-29T11:59:59+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[17799]: resource_api.cpp:190 Acquire port-config for ps3netsrv when 0x0002 (done)
2024-06-29T12:00:00+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_install[11011]: pkginstall.cpp:1221 Failed to acquire resource after install ps3netsrv [0xD900 manager.cpp:204]
2024-06-29T12:00:11+02:00 ##### synoscgi_SYNO.Core.Package_2_list[17933]: list.cpp:329 SYNO.Core.Package.list rollback to direct call
2024-06-29T12:00:45+02:00 ##### synoscgi_SYNO.Core.Package_2_list[20468]: list.cpp:329 SYNO.Core.Package.list rollback to direct call
```

That must be related to the package upgrade since a new installation on DSM 6.2.4 works.

REMARKS:
on DSM 6 the migration of etc/installer-variables is correct.

mreid-tt commented:
It's been a while since I've worked on this. Perhaps https://github.com/SynoCommunity/spksrc/pull/5962 may have some hints of how to resolve this?

hgy59 commented:
> It's been a while since I've worked on this. Perhaps #5962 may have some hints of how to resolve this?

@mreid-tt after testing with cops, I can tell that cops package provides a dialog in upgrade wizard for the share name, when installer-variables for share are not already populated.

For cops since 1.5.4-9 the installer variables SHARE_NAME and SHARE_PATH are added. until cops 1.4.2-7 the installer-variables file contained `GROUP=http` only.

On DSM 6.2.4 I was able to get the error `Failed to acquire resource after install cops` by adding `SHARE_NAME=calibre` to the installer-variables file before the update to v1.5.4-9.
When SHARE_NAME is defined, the upgrade wizard page for the share is not shown and I guess the error is caused by lacking variable `wizard_calibre_share`.

Further information on shared folder migration can be found in the discussions of #5968

### ps3netsrv on DSM 6.2.4:
This is really a mess:
After new installation of ps3netsrv v1.47.37-7 or v1.47.45-8 the installer-variables file contains one entry `SHARE_PATH=PS3` (the variable defines the share name as  `SHARE_PATH` and not `SHARE_NAME` !)
But it is even worse:
With an update of ps3netsrv from v1.47.37-7 or v1.47.45-8 the installer-variables file is removed.

For DSM 6 I propose to always show the upgrade wizard page for the share (or we have to correctly populate the variable `wizard_shared_folder_name` from `SHARE_PATH` variable since the current solution in cops does not solve this)

---
todo: analysis for DSM 7

hgy59 commented:
> The migration of `installer-variables` isn't correct. The upgrade might be missing something.
> 
> With the default share "PS3" the etc/installer-variables file looks like this:
> 
> former package version:
> 
> ```
> SHARE_PATH=PS3
> ```
> 
> new installation of new version:
> 
> ```
> SHARE_PATH=/volume1/PS3
> SHARE_NAME=PS3
> ```
> 

will be fixed with #6163

mreid-tt commented:
> @mreid-tt after testing with cops, I can tell that cops package provides a dialog in upgrade wizard for the share name, when installer-variables for share are not already populated.

Interesting, with the change you just made with https://github.com/SynoCommunity/spksrc/pull/6163, does it resolve the issue with COPS as well? Should a new version be published?

hgy59 commented:
@mreid-tt, @Hirador all installer issues are fixed now.

The upgrade wizard page for the rare case of unrecoverable share folder configuration is currently in english only.
If we want multiple languages we should migrate to wizard_templates...

hgy59 commented:
> > @mreid-tt after testing with cops, I can tell that cops package provides a dialog in upgrade wizard for the share name, when installer-variables for share are not already populated.
> 
> Interesting, with the change you just made with #6163, does it resolve the issue with COPS as well? Should a new version be published?

There is currently no issue in cops. 
I intentionally patched the installer-variables file to get an error.

The main reason that cops does not have this issue is the initialization of the `CONFIGURED_SHARE_NAME` variable from the package configuration and not from installer-variables.

But there might be an issue in sabnzb

Other packages with shared folders don't have an upgrade wizard (at least none with optional shared folder configuration) but I didn't hear about upgrade issues for those. so far..
- transmission
- gitea
- kiwix
- aria2
- mpd
- owncloud

Hirador commented:
> @mreid-tt, @Hirador all installer issues are fixed now.
> 
> 
> 
> The upgrade wizard page for the rare case of unrecoverable share folder configuration is currently in english only.
> 
> If we want multiple languages we should migrate to wizard_templates...

No worries, I was planning to just keep English for future releases. But If you wish me to migrate to wizard_templates anyway just give me an example and I'll try to figure it out as soon as I can

Hirador commented:
@hgy59 Let me know how we can proceed
